### ISSUE TITLE
Nginx sub-domain authentication

### ISSUE THREAD

**Issue by patrickmilnes** [NginxProxyManager/nginx-proxy-manager#1163]
**Checklist**
- Have you pulled and found the error with `jc21/nginx-proxy-manager:latest` docker image?
  - Yes
- Are you sure you're not using someone else's docker image?
  - Yes
- Have you searched for similar issues (both open and closed)?
  - Yes

**Describe the bug**

I have a web application and an authentication provider running in docker compose with nginx proxy manager. Im trying to add authentication, such that any atempt to access the web appplication is redirected via a sub-request to /auth endpoint of the authentication provider.

The custom config is being attached to the web app in advanced settings of the proxy host.

nginx custom conf:
```
auth_request    /auth;
auth_request_set $auth_status $upstream_status;

location = /auth {
  internal;
  proxy_pass              {endpoint/auth};
  proxy_pass_request_body off;
  proxy_set_header        Content-Length "";
}
```

**Nginx Proxy Manager Version**

v2.9.2

**To Reproduce**
1. Add above config to proxy host.
2. Navigate to proxy host.
3. Error

**Expected behavior**

If not authenticated, redirect to authentication provider login page. If authenticated, display proxy host.

**Screenshots**

![image](https://user-images.githubusercontent.com/33280737/121213773-72a45300-c876-11eb-8aea-3f81eea5df62.png)

**Operating System**

Docker on Ubuntu 18.04

**Additional context**

**Comment by LiamKarlMitchell** [comment]
Same on a new setup made today... seems like basic auth does not prompt the user?

**Comment by patrickmilnes** [comment]
@LiamKarlMitchell Yeah the prompt does not come up at all, just goes straight to the web page.

This should not have anything to do with access list.
